### DISCUSSION TITLE
Debugging existing and custom components

### DISCUSSION THREAD

**Discussion by codedearta** [langflow-ai/langflow#3379] [General]
When evaluating Langflow,I stept over a couple of hurdles. First, many components have very poor documentation, no examples how to use them. Secondly I ran into a couple of issues where a component didn't behave as I expected (Probably because I could not find detailed documentation). E.g. the "Directory" component is not really producing the expected list of Data. I wonder if someone could add some documentation on how to run Langflow in debugg mode so it can be attached to the pyhton debugger. That way developers could easily figure out themselves, what goes wrong with a certain component.

**Comment by codenprogressive** [comment]
Did you try to pass `--log-level debug` or in the env file: `LANGFLOW_LOG_LEVEL=debug`
doc: [https://docs.langflow.org/configuration-cli#4e811481ec9142f1b60309bb1ce5a2ce](https://docs.langflow.org/configuration-cli#4e811481ec9142f1b60309bb1ce5a2ce)

**Comment by SHQHDMR** [comment]
There are some issues with debugging custom components in Langflow. After examining the source code, I found that custom components are loaded into memory as code packages and then encapsulated into memory objects. These memory objects are located under /.../site-packages/langflow/utils/validate.py. This means that when we debug, we're actually executing the code from the objects in /.../site-packages/langflow/utils/validate.py (which contains our custom component code in the self._code field), rather than the code in our custom component directory.

I haven't found a good solution yet to make the debugger trace back to the original code in our custom component directory. When I step into the code, what I see is a bunch of Python-rewritten bytecode, which makes debugging quite inconvenient
